components/ai_chat/renderer/page_text_distilling.cc

```diff
@@ -204,4 +226,69 @@ void DistillPageText(
   std::move(callback).Run(contents_text);
 }
 
+void DistillPageTextViaSiteScript(
+    content::RenderFrame* render_frame,
+    std::string_view script_content,
+    int32_t world_id,
+    base::OnceCallback<void(const std::optional<std::string>&)> callback) {
+  // TODO (jonathansampson): Wrap scripts at build/transpile-time instead
```

Comment:
`CHECK(ai_chat::features::IsCustomSiteDistillerScriptsEnabled())`

Reply:
nit: `TODO(jonathansampson)` and also is there an issue you can link to?

Reply:
No issue to link to at this time; how we approach that might rely on whether or not the code is moved to a component.